quantum/process_keycode/process_unicode_common.c

```diff
@@ -52,17 +52,22 @@ void unicode_input_mode_init(void) {
     dprintf("Unicode input mode init to: %u\n", unicode_config.input_mode);
 }
 
-uint8_t get_unicode_input_mode(void) { return unicode_config.input_mode; }
+uint8_t get_unicode_input_mode(void) {
+    return unicode_config.input_mode;
+}
 
 void set_unicode_input_mode(uint8_t mode) {
     unicode_config.input_mode = mode;
     persist_unicode_input_mode();
     dprintf("Unicode input mode set to: %u\n", unicode_config.input_mode);
 }
 
-void cycle_unicode_input_mode(uint8_t offset) {
+void cycle_unicode_input_mode(int8_t offset) {
 #if UNICODE_SELECTED_MODES != -1
-    selected_index            = (selected_index + offset) % selected_count;
+    selected_index = (selected_index + offset) % selected_count;
+    if (selected_index < 0) {
+        selected_index += selected_count;
+    }
```

Comment:
Explanation:
First off, since you're supposed to be able to cycle backward (using a negative `offset`) as well as forward, the formal argument shouldn't be unsigned. The index calculation then tries to take `(selected_index + offset) mod selected_count`, but only succeeds partially in doing so since the C `%` operator is a remainder, not a modulo operation. In other words, the result can be negative, and if it is, it needs to be shifted back into the `[0, selected_count)` range.

The current behavior of this section (prior to the changes) is equivalent to the following pseudocode:
```
if offset < 0 then
    selected_index ← (selected_index + offset + 1) mod selected_count 
else
    selected_index ← (selected_index + offset) mod selected_count
fi
```
This is a quirk of the operands being unsigned, due to how two's complement works. Needless to say, it's not the desired behavior, and any arithmetic that expects negative operands should be changed to signed arithmetic.